src/main/java/com/salesforce/functions/jvm/sdk/data/bulk/BulkApi.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+package com.salesforce.functions.jvm.sdk.data.bulk;
+
+import com.salesforce.functions.jvm.sdk.data.error.DataApiException;
+import javax.annotation.Nonnull;
+
+public interface BulkApi {
+  /**
+   * Creates a new bulk ingest job.
+   *
+   * @param objectType The object type for the data being processed. Use only a single object type
+   *     per job.
+   * @param operation The processing operation for the job.
+   * @return A reference object to the created job.
+   * @throws DataApiException If an API error occurred during the operation.
+   * @see <a
+   *     href="https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/create_job.htm">API
+   *     Reference</a>
+   */
+  @Nonnull
+  @SuppressWarnings("unused")
+  IngestJobReference createIngestJob(String objectType, IngestOperation operation)
```

Comment:
Yeah, well spotted. I'm thinking how we can incorporate `externalIdFieldName` since it's only used for updates. Just slapping a new argument in there doesn't feel right to me.

Reply:
yeah, that's where i like a config object like `IngestJobOptions` that could be backed by a builder vs. making multiple overrides to support the optional arguments